### PR TITLE
jedi 2020-08-06 -> 0.18.0

### PR DESCRIPTION
**Pull request by breakds**
###### Motivation for this change

This is an attempt to bring `jedi` to its latest release. It is a dependency for a few of the other packages, which means that not upgrading jedi may be blocking the upgrade of others.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/jedi/default.nix [hunk 1]
```diff
@@ -1,34 +1,36 @@
...
-    rev = "216f976fd5cab7a460e5d287e853d11759251e52";
-    sha256 = "1kb2ajzigadl95pnwglg8fxz9cvpg9hx30hqqj91jkgrc7djdldj";
-    fetchSubmodules = true;
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-klUKQEutiv7YgaE37JpGH+1J7KZhQUvkUFkylhTtBwc=";
   };
 
-  checkInputs = [ pytest glibcLocales tox pytest-cov ];
+  checkInputs = [ glibcLocales tox pytestCheckHook ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/jedi/default.nix:19]:
```suggestion
  checkInputs = [ glibcLocales pytestCheckHook ];
```

File: pkgs/development/python-modules/jedi/default.nix [hunk 2]
```diff
@@ -1,34 +1,36 @@
...
 
   propagatedBuildInputs = [ parso ];
 
-  checkPhase = ''
-    LC_ALL="en_US.UTF-8" py.test test
-  '';
-
-  # tox required for tests: https://github.com/davidhalter/jedi/issues/808
+  # NOTE(breakds): There are 21 tests among ~ 4000 of them failing at
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/jedi/default.nix:23]: Can we just disable the 21 ones instead?

Comment (breakds) [reply pkgs/development/python-modules/jedi/default.nix:23]: Sure, will try to disable them precisely. I am running into other problems that this PR breaks some other packages. Let me try to fix them.

**Comment by doronbehar** [comment]
jedi is updated now.

**Pull request closed without merging**